From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Fri, 18 Oct 2024 17:01:06 +0200
Subject: fix: remove outdated V8 flags from node.cc

Refs https://chromium-review.googlesource.com/c/v8/v8/+/5507047
Refs https://chromium-review.googlesource.com/c/v8/v8/+/6249026
Refs https://chromium-review.googlesource.com/c/v8/v8/+/6948286

The above CL removes the `--harmony-import-assertions`,
`--experimental-wasm-imported-strings`, and
`--experimental-wasm-memory64` flags from V8.

This patch can be removed when we upgrade to a V8 version that
contains the above CLs.

diff --git a/src/node.cc b/src/node.cc
index 2d8d0000d52411992d2bd513cc7dd96b2292bab9..0ffc69d54631fa3dbac20337b26f7a0679560b96 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -816,7 +816,7 @@ static ExitCode ProcessGlobalArgsInternal(std::vector<std::string>* args,
   }
   // TODO(nicolo-ribaudo): remove this once V8 doesn't enable it by default
   // anymore.
-  v8_args.emplace_back("--no-harmony-import-assertions");
+  // v8_args.emplace_back("--no-harmony-import-assertions");
 
   if (!per_process::cli_options->per_isolate->max_old_space_size_percentage
            .empty()) {
@@ -834,8 +834,8 @@ static ExitCode ProcessGlobalArgsInternal(std::vector<std::string>* args,
   }
 
   // Support stable Phase 5 WebAssembly proposals
-  v8_args.emplace_back("--experimental-wasm-imported-strings");
-  v8_args.emplace_back("--experimental-wasm-memory64");
+  //v8_args.emplace_back("--experimental-wasm-imported-strings");
+  // v8_args.emplace_back("--experimental-wasm-memory64");
   v8_args.emplace_back("--experimental-wasm-exnref");
 
 #ifdef __POSIX__
